src/eal.cpp

```diff
@@ -68,13 +67,54 @@ namespace eal {
   }
 
   // Postion Function Functions
-  vector<double> getPosition(vector<double> coeffs, double time) {
+  // vector<double> coeffs = [[cx_n, cx_n-1, cx_n-2, ... cx_0],
+  //                          [cy_n, cy_n-1, cy_n-2, ... cy_0],
+  //                          [cz_n, cz_n-1, cz_n-2, ... cz_0]]
+  // The equations evaluated by this function are:
+  //                x = cx_n * t^n + cx_n-1 * t^(n-1) + ... + cx_0
+  //                y = cy_n * t^n + cy_n-1 * t^(n-1) + ... + cy_0
+  //                z = cz_n * t^n + cz_n-1 * t^(n-1) + ... + cz_0
+  vector<double> getPosition(vector<vector<double>> coeffs, double time) {
+    if ( time < 0.0) {
+      throw invalid_argument("Invalid input time, must be non-negative.");
+    }
+
+    if (coeffs.size() != 3) {
+      throw invalid_argument("Invalid input coeffs, expected three vectors.");
+    }
+
+    // make sure all coeffs sizes are equal, else throw error...
+    if ((coeffs[0].size() != coeffs[1].size()) || 
+        (coeffs[1].size() != coeffs[2].size())) {
+      throw invalid_argument("Invalid input coeffs, each must represent the same degree polynomial.");
```

Comment:
This is kind of arbitrary, but if you had a 2nd degree solution for x, but 4 for z, and 0 for y, we could say: use the highest degree solution to pick the size, and then put in 0s for everything else.

Reply:
I think we can allow for any size, just require that they are all non-empty

Reply:
:+1: will do